1525045: Simplify datetime to just have value, not unit

mdboom opened the pull request:
Checklist for reviewer:

- [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] Trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional)

fbertsch commented:
@mdboom this is not a schema-compatible change :/ we'll need to bump the version. Sound good?

mdboom commented:
Sure thing.  To confirm: this would involve copying the `datetime` template and all of the downstream (datetime-using) templates to version 2 (and leaving version 1 as it was prior to this change)?  Or do I need to bump the version on all the glean templates?

fbertsch commented:
@mdboom exactly. So old clients will route to JSON schema v1, and then parquet tables v1; new clients will need to bump their docVersion to hit v2 of both. On the query side we'll need to include both tables to get client counts (depending on how long it takes everyone to update, which given the reference browser population, won't be long).

mdboom commented:
Cool.  I've redone this so it's all as a bump to version 2.

fbertsch commented:
Big fan of the changelog. Consider adding something to the README about it.

Should we update the changelog for schema-compatible changes?

mdboom commented:
> Big fan of the changelog. Consider adding something to the README about it.

Would you just suggest something like "For glean schemas, we have a changelog here...", or are you suggesting a central changelog or "everybody gets a changelog"?

> Should we update the changelog for schema-compatible changes?

That's probably a good idea, too.

Dexterp37 commented on validation/glean/baseline.2.core.pass.json:
```diff
@@ -0,0 +1,97 @@
+{
+  "$schema": "moz://mozilla.org/schemas/glean/ping/1",
```
Quick flyby (the rest looks good to me): should this end with `/2`, given the version bump? Or it doesn't matter/is something else?

mdboom replied:
Good catch.  You are right, this should be updated.

fbertsch commented:
For now, add a bit about generally how to use the changelog, and then point to the glean one in particular. In the future we can roll it out to other pings if we feel like it's been useful here.

We should probably require updates to the changelog for glean ping updates.

mdboom commented:
Cool.  I've updated the main `README` about the glean `CHANGELOG`.  I also updated the pull request template -- let me know if that's too noisy, since many (most) PRs probably don't need that.
